engine/include/Engine/Intermediate/GameObject.hpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include "Engine/Intermediate/TransformComponent.hpp" //TransformComponent
+#include <list>                                       //std::list
+#include <memory>                                     //std::unique_ptr
+#include <string>                                     //std::string
+#include <vector>                                     //std::vector
+
+// in Inl
+#include "Engine/Intermediate/DataChunk.hpp"
+
+namespace GPE
+{
+class Component;
+
+class GameObject
+{
+public:
+    struct CreateArg
+    {
+        std::string                   name = "";
+        TransformComponent::CreateArg transformArg;
+    };
+
+protected:
+    std::string         m_name;
+    TransformComponent* m_pTransform;
```

Comment:
**Suggestion**

Make `m_pTransform` a reference to TransformComponent, or don't leave it as a pointer but don't initialize it as long as a transform is not needed